python/ray/_private/accelerators/tpu.py

```diff
@@ -247,3 +191,187 @@ def set_current_process_visible_accelerator_ids(
+        if not accelerator_type:
+            # GCE-based VM check
+            accelerator_type = _get_tpu_metadata(key=GCE_TPU_ACCELERATOR_KEY)
+        if TPUAcceleratorManager.is_valid_tpu_accelerator_type(
+            tpu_accelerator_type=accelerator_type
+        ):
+            return accelerator_type
+        logging.debug("Failed to get a valid accelerator type.")
+        return None
+
+    @staticmethod
+    def get_tpu_id() -> Optional[str]:
+        """Return the name of the TPU pod that this worker node is a part of.
+
+        For instance, if the TPU was created with name "my-tpu", this function
+        will return "my-tpu".
+
+        If created through the Ray cluster launcher, the
+        name will typically be something like "ray-my-tpu-cluster-worker-aa946781-tpu".
+
+        In case the TPU was created through KubeRay, we currently expect that the
+        environment variable TPU_ID is set per TPU pod slice, in which case
+        this function will return the value of that environment variable.
+
+        """
+        try:
+            # Start with GKE-based check
+            tpu_id = os.getenv(GKE_TPU_ID_ENV_VAR, None)
+            if not tpu_id:
+                # GCE-based VM check
+                tpu_id = _get_tpu_metadata(key=GCE_TPU_INSTANCE_ID_KEY)
+            return tpu_id
+        except ValueError as e:
+            logging.debug("Could not get TPU worker id: %s", e)
+            return None
+
+    @staticmethod
+    def _get_tpu_worker_id() -> Optional[int]:
+        """Return the worker index of the TPU pod."""
+        try:
+            # Start with GKE-based check
+            worker_id = os.getenv(GKE_TPU_WORKER_ID_ENV_VAR, None)
+            if not worker_id:
+                # GCE-based VM check
+                worker_id = _get_tpu_metadata(key=GCE_TPU_WORKER_ID_KEY)
+            return int(worker_id)
+        except ValueError as e:
+            logging.debug("Could not get TPU worker id: %s", e)
+            return None
+
+    @staticmethod
+    def num_workers_in_tpu_pod() -> Optional[int]:
+        """Return the total number of workers in a TPU pod."""
+        accelerator_type = TPUAcceleratorManager._get_tpu_accelerator_type()
+        if accelerator_type:
+            version = accelerator_type.split("-")[0]
+            num_chips_or_cores = int(accelerator_type.split("-")[1])
+            if version in TPU_VERSIONS_WITH_MULTIPLE_CORES_PER_CHIP:
+                return num_chips_or_cores // 8
+            else:
+                return num_chips_or_cores // 4
+        else:
+            logging.debug("Could not get num workers in TPU pod.")
+            return None
+
+    @staticmethod
+    def get_current_node_accelerator_type() -> Optional[str]:
+        """Attempt to detect the TPU accelerator type.
+
+        Returns:
+            A string representing the TPU accelerator type,
+            e.g. "TPU-V2", "TPU-V3", "TPU-V4" if applicable, else None.
+
+        """
```

Comment:
Also add more comments about these two different accelerator types and when to use which as we have discussed.

Reply:
Done